plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XBuildManager.java

```diff
@@ -302,52 +306,30 @@ protected void queue(Set<URI> files, Collection<URI> toRemove, Collection<URI> t
 
 	/** @since 2.18 */
 	protected void mergeWithUnreportedDeltas(List<IResourceDescription.Delta> newDeltas) {
-		if (this.unreportedDeltas.isEmpty()) {
-			unreportedDeltas.addAll(newDeltas);
+		if (this.allBuildDeltas.isEmpty()) {
+			allBuildDeltas.addAll(newDeltas);
 		} else {
 			Map<URI, IResourceDescription.Delta> unreportedByUri = IterableExtensions.toMap(
-					unreportedDeltas, IResourceDescription.Delta::getUri);
+					allBuildDeltas, IResourceDescription.Delta::getUri);
 
 			for (IResourceDescription.Delta newDelta : newDeltas) {
 				IResourceDescription.Delta unreportedDelta = unreportedByUri.get(newDelta.getUri());
 				if (unreportedDelta == null) {
-					unreportedDeltas.add(newDelta);
+					allBuildDeltas.add(newDelta);
 				} else {
-					unreportedDeltas.remove(unreportedDelta);
+					allBuildDeltas.remove(unreportedDelta);
 					IResourceDescription _old = unreportedDelta.getOld();
 					IResourceDescription _new = newDelta.getNew();
-					unreportedDeltas.add(new DefaultResourceDescriptionDelta(_old, _new));
+					allBuildDeltas.add(new DefaultResourceDescriptionDelta(_old, _new));
 				}
 			}
 		}
 	}
 
-	/** Get a sorted list of projects to be build. */
-	protected List<ProjectDescription> sortByDependencies(Iterable<ProjectDescription> projectDescriptions) {
-		List<ProjectDescription> sortedProjectDescriptions = topoSorter.sortByDependencies(projectDescriptions,
-				this::reportDependencyCycle);
-
-		String output = "Project build order:\n  ";
-		output += String.join("\n  ", IterableExtensions.map(sortedProjectDescriptions, pd -> pd.getName()));
+	/** Prints build order */
+	protected void printBuildOrder(IOrderInfo<ProjectDescription> orderInfo) {
+		String output = "Project build order:\n  "
```

Comment:
I know that this is "just" a string concatenation, but please surround this by `Log.isInfoEnabled()`